lib/gltfPipeline.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+var fs = require('fs');
+var path = require('path');
+var addDefaults = require('./addDefaults');
+var removeUnusedImages = require('./removeUnusedImages');
+var removeUnusedSamplers = require('./removeUnusedSamplers');
+var removeUnusedShaders = require('./removeUnusedShaders');
+var removeUnusedTechniques = require('./removeUnusedTechniques');
+var removeUnusedPrograms = require('./removeUnusedPrograms');
+var removeUnusedBuffers = require('./removeUnusedBuffers');
+var removeUnusedBufferViews = require('./removeUnusedBufferViews');
+var removeUnusedMaterials = require('./removeUnusedMaterials');
+var removeUnusedSkins = require('./removeUnusedSkins');
+var removeUnusedCameras = require('./removeUnusedCameras');
+var removeUnusedTextures = require('./removeUnusedTextures');
+var removeUnusedMeshes = require('./removeUnusedMeshes');
+var removeUnusedNodes = require('./removeUnusedNodes');
+var removeUnusedAccessors = require('./removeUnusedAccessors');
+var removeUnused = require('./removeUnused');
+var loadGltfUris = require('./loadGltfUris');
+var parseBinaryGltf = require('./parseBinaryGltf');
+var addPipelineExtras = require('./addPipelineExtras');
+var convertDagToTree = require('./convertDagToTree');
+var combineMeshes = require('./combineMeshes');
+var combinePrimitives = require('./combinePrimitives');
+var removeUnusedVertices = require('./removeUnusedVertices');
+var OptimizationStatistics = require('./OptimizationStatistics');
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+var defaultValue = Cesium.defaultValue;
+var DeveloperError = Cesium.DeveloperError;
+
+module.exports = gltfPipeline;
+
+function gltfPipeline(options, callback) {
+    var gltfPath = options.inputPath;
+    
+    if (!defined(gltfPath)) {
+        throw new DeveloperError('Input path is undefined.');
+    }
+    
+    var fileExtension = path.extname(gltfPath);
+    var fileName = path.basename(gltfPath, fileExtension);
+    var filePath = path.dirname(gltfPath);
+
+    if (fileExtension !== '.glb' && fileExtension !== '.gltf') {
+        throw new Error('Invalid glTF file.');
+    }
+
+    fs.readFile(gltfPath, function (err, data) {
+        if (err) {
+            throw err;
+        }
+
+        var gltf;
+        if (fileExtension === '.glb') {
+            gltf = parseBinaryGltf(data);
+        }
+        else if (fileExtension === '.gltf') {
+            gltf = JSON.parse(data);
+            addPipelineExtras(gltf);
+        }
+
+        var stats = new OptimizationStatistics();
+
+        addDefaults(gltf, stats);
+
+        removeUnused(gltf, stats);
+        printStats(stats);
+
+        convertDagToTree(gltf);
+
+        gltf = loadGltfUris(gltf, filePath, function (err) {
+            if (err) {
+                throw err;
+            }
+
+            combineMeshes(gltf);
+            combinePrimitives(gltf);
+
+            removeUnusedVertices(gltf);
+
```

Comment:
I was actually thinking about moving removeUnusedVertices to come before the combine meshes/primitives/nodes stage, so that the actual scene data would be optimized before condensing the gltf structure.